command/agent/retry_join_test.go

```diff
@@ -0,0 +1,171 @@
+package agent
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/hashicorp/nomad/version"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+)
+
+type MockDiscover struct{}
```

Comment:
You could make this struct take a result address list as a member variable and return that in the Addrs method below. That way the _Cloud test can set up the mock provider to hardcode an IP and verify that it returns that as opposed to returning the string passed to it the `Addrs` method